### PR TITLE
Remove an unnecessary corner case

### PR DESCRIPTION
**Pull request by wujingyue**

### CODE REVIEW

**Comment by wujingyue** [comment]
!test

**Comment by wujingyue** [comment]
!test

**Review by jjsjann123 — APPROVED** [review]

File: csrc/scheduler/vectorize_helper.cpp [hunk 1]
```diff
@@ -760,11 +760,6 @@ Val* ContiguousInnerDimensionsMapper::getContigMergeOfInnerSize(
 
   auto alloc_no_reductions_size = alloc_no_reductions.size();
 
-  // Filter out 0-dim tensors
-  if (alloc_no_reductions_size < 1) {
-    return product_of_inner_extents;
-  }
-
```
Comment (jjsjann123) [inline csrc/scheduler/vectorize_helper.cpp:767]: looks like the assertion below still got hit. So we would want to change that assert to conditional instead?

Comment (wujingyue) [reply csrc/scheduler/vectorize_helper.cpp:767]: What do you mean? Which assertion got hit? None of the CI errors seem to be related at all.

Comment (jjsjann123) [reply csrc/scheduler/vectorize_helper.cpp:767]: ahh... You are right, I must have mis-read the thunder assert. 😮‍💨 

> 00:23:06 >       torch.testing.assert_close(actual_fx, expected)
00:23:06 E       AssertionError: The values for attribute 'shape' do not match: torch.Size([]) != torch.Size([2]).
